### ISSUE TITLE
Drag and Drop completely broken in ChromeOS/Android/iOS/iPadOS

### ISSUE THREAD

**Issue by MarsOrBust24** [stackblitz/core#1762]
# Tl;Dr:

Please insert `.filter(e => e !== null )` after `webkitGetAsEntry())` in the drag and drop, drop handling function.

---

### **Description of Bug**
Cannot drag and drop anything at all. I will get the upload hand when I have a single file, and the error hand when I hover with a folder.

### **Steps to Reproduce**

1. Drag and Drop with any version of ChromeOS. Tested on an HP 14c

### **Expected Behavior**
Drag and drop should work without subprocessing with additional third parties, and those who are required to use Gitlab, Bitbucket, on-prem etc. Github is also cumbersome to use if your CI/git ci/k8s pipeline is handled by another vendor.

### **Screenshots/Screencast**
![Screenshot 2022-02-25 23 41 59](https://user-images.githubusercontent.com/876076/155831028-40b8fed6-efeb-4b55-95b9-776e98951a62.png)

# Proposed solution

Filter out null return values from `webkitGetAsEntry`, https://github.com/stackblitz/core/issues/1762#issuecomment-1170626502

# Additional info
- It works in Ubuntu...
- This code and Sandbox IDE also work https://codesandbox.io/s/github/react-dnd/react-dnd/tree/gh-pages/examples_ts/06-other/native-files

## Workarounds

- Hijack Runkit to run a cors-proxy and then use isomorphic-git to download stuff
- Make a temporary github repository, copy all your code into it, upload your files to it, then connect it to SB, then disconnect.
- run a mutter file upload service
- If you have a Chromebook with 16GB of ram or more... ie, you are a Google developer, School administrator or a Google Workspace business Executive, see https://github.com/stackblitz/core/issues/1762#issuecomment-1170626502

**Comment by rayfoss** [comment]
Using a good useragent extensions to pretend to be windows did not help.

The issue is you're checking in wP(e) if e.isFile, problem is the second time wP runs after a drag and drop, e is null. Using `e?.isFile` should fix it.

here's the first time it runs:
![Screenshot 2022-03-03 18 45 48](https://user-images.githubusercontent.com/876076/156679091-a9a61e09-9a77-4614-b337-bae597812441.png)

**Comment by rayfoss** [comment]
Is there an upload button or something I'm missing?

The only workaround I can think of is hosting a server on the local machine and connecting to it from Web Containers. Only issue is the server has to use an HTTP protocol, so FTP and SSH are out of the picture, right? I see the ssh2 package has an HTTPAgent, or is WebDAV viable?

This is a huge issue. Starting projects from a Zip file would be great for migrations from CodeSandbox.

**Comment by tdpowerskills** [comment]
Same issue for months... seems like a quick fix and very easy to reproduce, nothing can be uploaded to Sandboxes from ChromeOS.

Please remove the triage label and copy to internal tracker
![Screenshot 2022-04-15 11 02 52 AM](https://user-images.githubusercontent.com/87497159/163593620-ffda86ca-6d2e-4ac6-a265-014f687eaad1.png)

**Comment by rayfoss** [comment]
Please fix... this makes SB extremely difficult to use on ChromeOS... If you have a spare intel laptop you can test on ChromeOS Flex. ChromeOS tablets can go for less than $100.

I think I found the problem:
![Screenshot 2022-06-23 19 31 39](https://user-images.githubusercontent.com/876076/175437270-7a204b9e-fd21-4774-a2cb-c7c16a6bb2eb.png)

**Comment by rayfoss** [comment]
https://gs.statcounter.com/os-market-share/desktop/united-states-of-america/#monthly-202103-202203

For trage purposes, this affects schools and junior developers disproportionately and makes StackBlitz near unusable in the platform. Installing other browsers on ChromeOS's Crostini Linux environment doesn't fix the drag and drop bug either.

This will mean StackBlitz will be unusable to about 10% of potential customers including large educational institutions come this August school season.

**Comment by rayfoss** [comment]
# Update #

I just barely managed to hack in this patch on ChromeOS using the Chrome DevTools overrides... this allows ChromeOS users to drag and drop single and multiple files, but not folders. This would be a good enough fix for now...

Unfortunately, using overrides to sideload a deoptimized modified version of client.js takes so much ram that my 8GB Chromebook will SIGSEGV crash 85% of the time, with no other tabs open. Higher RAM capacity like the HP Dragonfly Chromebook, C640 Enterprise Chromebook and high spec Chromebook Pixel's are rare.

```js
{
// This is in line 134255 of client.js in c.stackblitz.com
drop: (e, t) => {
                    switch (t.getItemType()) {
                      case BI.FILE: {
                        // <MODIFICATION>
                        const eRaw = [...t.getItem().items].map((e) =>
                            e.webkitGetAsEntry()
                          );
                        const e = eRaw.filter(e => e !== null ); // Filter out null values
                        const n = e.map(WI);
                        // </MODIFICATION>
                        o(
                          y || "",
                          Promise.all(n).then((e) => e.flatMap((e) => e))
                        );
                        break;
                      }
                      case GZ:
                        r(
                          e.file,
                          "folder" === g.type ? g.fullPath : sZ(g.fullPath)
                        );
                    }
                  }
}
```

**Comment by rayfoss** [comment]
The ReactDnD version works as expected...

https://react-dnd.github.io/react-dnd/examples/other/native-files

**Comment by MarsOrBust24** [comment]
🙏
